File: 3_longestSubstring.js
function lengthOfLongestSubstring(s){
    let left=0;
    let maxLen=0;
    let seen = new Set();

    for(let right=0; right<s.length; right++)
    {
        while(seen.has(s[right]))
        {
            seen.delete(s[right]);
            left++;
        }

        seen.add(s[right]);
        maxLen= Math.max(maxLen,right-left+1);
    }

    return maxLen;
}

let s='abcdeffff';
console.log(lengthOfLongestSubstring(s));             

File: Coding_Problems/CPP/3_longestSubString.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <algorithm>

using namespace std;

int lengthOfLongestSubstring(string s) {

    int left = 0;
    int maxLen = 0;

    unordered_set<char> seen;

    for (int right = 0; right < s.size(); right++) {

        while (seen.count(s[right])) {
            seen.erase(s[left]);
            left++;
        }

        seen.insert(s[right]);

        maxLen = max(maxLen, right - left + 1);
    }

    return maxLen;
}

int main() {

    string s = "abcabcbbabcdef";

    cout << lengthOfLongestSubstring(s);

    return 0;
}